File: cpp/80_remove_duplicates_from_sorted_array_II.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int count = 0;
        for(int i = 0; i < nums.size(); i++){
            int k = i + 1;
            while(k < nums.size() && nums[k] == nums[i])
                k++;

            if(k == i + 1){
                nums[count] = nums[i];
                count++;
                continue;
            }

            nums[count] = nums[i], count++;
            nums[count] = nums[i], count++;

            i = k -1;
        }
        return count;
    }
};

File: rs/80_remove_duplicates_from_sorted_array_II.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut count = 0;
        let mut i = 0;

        while i < nums.len() {
            let mut k = i + 1;
            while k < nums.len() && nums[k] == nums[i] {
                k += 1;
            }

            if k == i + 1 {
                nums[count] = nums[i];
                count += 1;
            } else {
                nums[count] = nums[i];
                count += 1;
                nums[count] = nums[i];
                count += 1;
            }

            i = k;
        }

        count as i32
    }
}